src/main/java/com/socialmedia/service/AmazonClient.java

```diff
@@ -0,0 +1,80 @@
+package com.socialmedia.service;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+@Service
+@Configuration
+@PropertySource("classpath:/amazonKeys/keys-for-s3.properties")
+public class AmazonClient {
+
+    private AmazonS3 s3client;
+
+    @Value("${amazonProperties.endpointUrl}")
+    private String endpointUrl;
+    @Value("${amazonProperties.bucketName}")
+    private String bucketName;
+    @Value("${secretKey.accessKey}")
+    private String accessKey;
+    @Value("${secretKey.secretKey}")
+    private String secretKey;
+
+    @PostConstruct
+    private void initializeAmazon() {
```

Comment:
acceptable, but it would be better to create amazonClient as bean in configuration and inject it to this class